### PR TITLE
[MER-1694] Anonymous posts for students

### PR DESCRIPTION
**Pull request by nicocirio**
## Story

[MER-1694](https://eliterate.atlassian.net/browse/MER-1694)

## Notes

* Instead of having a checkbox to mark a post as anonymous, I decided a button with options (similar to the ones we have in github) would improve the user experience and design

https://user-images.githubusercontent.com/74839302/222527146-63c9133b-9218-4fa7-aa2f-cefee776104b.mov

https://user-images.githubusercontent.com/74839302/222532603-3b68729f-cccb-4ab6-b44d-a9115c8fd3b3.mov

* This PR includes the restyling of the CollabSpaceConfigView, done with @augusto1024 and @simonchoxx. Since there was no design provided in Figma, we intended to respect the look&feel of Figma's designs

     * Before

       before

     * After
      after

## TODO
- [x] Fix failing test
- [x] Add tests

[MER-1694]: https://eliterate.atlassian.net/browse/MER-1694?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/oli/sections_test.exs [hunk 1]
```diff
@@ -1125,6 +1127,65 @@ defmodule Oli.SectionsTest do
...
+      map = Seeder.base_project_with_resource2()
+
+      institution = Map.get(map, :institution)
+      project = Map.get(map, :project)
+
+      valid_attrs =
+        Map.put(@valid_attrs, :institution_id, institution.id)
+        |> Map.put(:base_project_id, project.id)
+
+      user = user_fixture()
```
Comment (simonchoxx) [inline test/oli/sections_test.exs]: I think that the user's factory function (`user = insert(:user)`) could have been used here as well

File: test/oli/sections_test.exs [hunk 2]
```diff
@@ -1125,6 +1127,65 @@ defmodule Oli.SectionsTest do
...
+
+      {:ok, section} = valid_attrs |> Sections.create_section()
```
Comment (simonchoxx) [inline test/oli/sections_test.exs]: The same as with the user, I think you could use the factory module function to insert a function, passing it the attributes that are needed

Something like... `section = insert(:section, @valid_attrs)`

**Review by simonchoxx — COMMENTED** [review]
Nice work mate!! 🚀🚀 

I just left you some comments, but no need to change them!

**Comment by nicocirio** [comment]
closed in favor or [PR opened in oli-torus](https://github.com/Simon-Initiative/oli-torus/pull/3303)

**Pull request closed without merging**